src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -98,192 +89,270 @@ private InventoryUtils()
+    {
+        return filterItemHandler(itemHandler, (ItemStack stack) -> compareItems(stack, targetItem, itemDamage));
+    }
+
+    /**
+     * Returns the index of the first occurrence of the block in the {@link
+     * IItemHandler}.
+     *
+     * @param itemHandler {@link IItemHandler} to check.
+     * @param block       Block to find.
+     * @param itemDamage  the damage value.
      * @return Index of the first occurrence.
      */
-    public static int findFirstSlotInInventoryWith(@NotNull final IInventory inventory, final Block block, int itemDamage)
+    public static int findFirstSlotInItemHandlerWith(@NotNull final IItemHandler itemHandler, @NotNull final Block block, int itemDamage)
     {
-        return findFirstSlotInInventoryWith(inventory, getItemFromBlock(block), itemDamage);
+        return findFirstSlotInItemHandlerWith(itemHandler, getItemFromBlock(block), itemDamage);
     }
 
     /**
-     * {@link #findFirstSlotInInventoryWith(IInventory, Block, int)}.
+     * Returns the index of the first occurrence of the Item with the given
+     * ItemDamage in the {@link IItemHandler}.
      *
-     * @param inventory  Inventory to check
-     * @param targetItem Item to find
-     * @param itemDamage the damage value.
+     * @param itemHandler {@link IItemHandler} to check
+     * @param targetItem  Item to find.
+     * @param itemDamage  The damage value to match on the stack.
+     * @return Index of the first occurrence
+     */
+    public static int findFirstSlotInItemHandlerWith(@NotNull final IItemHandler itemHandler, @NotNull final Item targetItem, int itemDamage)
+    {
+        return findFirstSlotInItemHandlerWith(itemHandler, (ItemStack stack) -> compareItems(stack, targetItem, itemDamage));
+    }
+
+    /**
+     * Returns the index of the first occurrence of an ItemStack that matches
+     * the given predicate in the {@link IItemHandler}.
+     *
+     * @param itemHandler                 ItemHandler to check
+     * @param itemStackSelectionPredicate The predicate to match.
      * @return Index of the first occurrence
      */
-    public static int findFirstSlotInInventoryWith(@NotNull final IInventory inventory, final Item targetItem, int itemDamage)
+    public static int findFirstSlotInItemHandlerWith(@NotNull final IItemHandler itemHandler, @NotNull final Predicate<ItemStack> itemStackSelectionPredicate)
     {
-        for (int slot = 0; slot < inventory.getSizeInventory(); slot++)
+        for (int slot = 0; slot < itemHandler.getSlots(); slot++)
         {
-            if (compareItems(inventory.getStackInSlot(slot), targetItem, itemDamage))
+            if (itemStackSelectionPredicate.test(itemHandler.getStackInSlot(slot)))
             {
                 return slot;
             }
         }
+
         return -1;
         //TODO: Later harden contract to remove compare on slot := -1
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)